graph_model/trie_model/trie_node.py

```diff
@@ -1,8 +1,5 @@
-from numpy import char
-
 class TrieNode:
-    def __init__(self, prefix = '', char = '', parent = None):
-        self.parent = parent
+    def __init__(self, prefix = '', char = ''):
```

Comment:
```suggestion
    def __init__(self, char:str, score:int=0, children:dict={}):
```
This might make the structure a bit simpler.